test/doc/core_snippets.cpp

```diff
@@ -38,7 +38,9 @@ namespace ssl = boost::asio::ssl;
 using tcp = net::ip::tcp;
 
 net::io_context ioc;
-auto work = net::make_work_guard(ioc);
+auto work = net::any_io_executor(
```

Comment:
This is weird, we need to put a pin in this and determine if it is what we really want to teach, as it shows up in the docs.